### PR TITLE
LPS-116501 Add Sidebar tests

### PR DESCRIPTION
**Pull request by dgarciasarai**
**Description**

Add jest tests for Sidebar component

cc: @carloslancha

### CODE REVIEW

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:relevant

**Comment by dgarciasarai** [comment]
ci:stop

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:relevant

File: modules/apps/content-dashboard/content-dashboard-web/test/js/components/Sidebar.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	beforeEach(() => {
+		jest.useFakeTimers();
+	});
+
+	afterEach(() => {
+		jest.clearAllTimers();
+		cleanup();
+	});
+
+	it('renders sidebar open', () => {
```
Comment (carloslancha) [inline modules/apps/content-dashboard/content-dashboard-web/test/js/components/Sidebar.js:43]: "renders sidebar open" -> "renders a default sidebar" or "renders an open sidebar"

File: modules/apps/content-dashboard/content-dashboard-web/test/js/components/Sidebar.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		render(<Sidebar />);
+
+		act(() => {
+			jest.runAllTimers();
+		});
+
+		expect(document.body).toHaveAttribute('class', 'sidebar-open');
+	});
+
+	it('renders sidebar close', () => {
```
Comment (carloslancha) [inline modules/apps/content-dashboard/content-dashboard-web/test/js/components/Sidebar.js:53]: "renders sidebar close" -> "renders a closed sidebar"

File: modules/apps/content-dashboard/content-dashboard-web/test/js/components/Sidebar.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+		render(<Sidebar open={false} />);
+
+		act(() => {
+			jest.runAllTimers();
+		});
+
+		expect(document.body).not.toHaveAttribute('class', 'sidebar-open');
+	});
+
+	it('renders sidebar closing', () => {
```
Comment (carloslancha) [inline modules/apps/content-dashboard/content-dashboard-web/test/js/components/Sidebar.js:63]: "renders sidebar closing" -> "renders an open sidebar and close it on close button click"

File: modules/apps/content-dashboard/content-dashboard-web/test/js/components/Sidebar.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+		expect(closeIcon).toBeInTheDocument();
+
+		userEvent.click(closeIcon);
+
+		act(() => {
+			jest.runAllTimers();
+		});
+
+		expect(document.body).not.toHaveAttribute('class', 'sidebar-open');
```
Comment (carloslancha) [inline modules/apps/content-dashboard/content-dashboard-web/test/js/components/Sidebar.js:87]: Are we maybe testing here things  not related with the case (that title and subtitle are being rendered, sidebar-open is in the document, closeIcon is in the document...)?

I'd suggest to create other case(s) where you test if the SidebarHeader are being rendered inside the sidebar, and maybe other for SidebarBody, and leave this case just for:

- Render open sidebar
- Click close button
- Check is closed

File: modules/apps/content-dashboard/content-dashboard-web/test/js/components/Sidebar.js (hunk 1 above)
Comment (dgarciasarai) [reply modules/apps/content-dashboard/content-dashboard-web/test/js/components/Sidebar.js:43]: 👍

File: modules/apps/content-dashboard/content-dashboard-web/test/js/components/Sidebar.js (hunk 2 above)
Comment (dgarciasarai) [reply modules/apps/content-dashboard/content-dashboard-web/test/js/components/Sidebar.js:53]: 👍

File: modules/apps/content-dashboard/content-dashboard-web/test/js/components/Sidebar.js (hunk 3 above)
Comment (dgarciasarai) [reply modules/apps/content-dashboard/content-dashboard-web/test/js/components/Sidebar.js:63]: 👍

File: modules/apps/content-dashboard/content-dashboard-web/test/js/components/Sidebar.js (hunk 4 above)
Comment (dgarciasarai) [reply modules/apps/content-dashboard/content-dashboard-web/test/js/components/Sidebar.js:87]: Ok, I'm going to update the existing test and create those new ones that you mentioned. Thanks!

**Comment by dgarciasarai** [comment]
@carloslancha I just update the PR with your feedback. Please, review it and if everything looks good, run the CI 😄 Thanks!!

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by carloslancha** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by carloslancha** [comment]
Resent https://github.com/liferay-frontend/liferay-portal/pull/184 applying some minor changes 😉

**Pull request closed without merging**